### PR TITLE
fish: set correct completions dir

### PR DESCRIPTION
**Pull request by xyb**
to fix broken fish completion support

### CODE REVIEW

File: Library/Formula/fish.rb [hunk 1]
```diff
@@ -21,6 +21,9 @@ class Fish < Formula
 
   def install
     system "autoconf" if build.head?
+    inreplace "fish.pc.in" do |s|
+      s.gsub! /^completionsdir=.*$/, "completionsdir=#{HOMEBREW_PREFIX}/share/fish/vendor_completions.d"
```
Comment (mikemcquaid) [inline Library/Formula/fish.rb]: Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

Comment (xyb) [inline Library/Formula/fish.rb]: Fish [release 2.2.0](https://github.com/fish-shell/fish-shell/releases/tag/2.2.0) said:

> Fish now defines a default directory for other packages to provide completions. By default this is `/usr/share/fish/vendor-completions.d`; on systems with `pkgconfig` installed this path is discoverable with `pkg-config --variable completionsdir fish`.

Follow fish's guide, this patch customized fish completions dir by modify `fish.pc` to meet homebrew's directory requirment. In other words, let fish not use fish's default `/usr/share/fish/vendor_completions.d` but homebrew's `/usr/local/share/fish/vendor_completions.d`.

Comment (xyb) [inline Library/Formula/fish.rb]: @mikemcquaid so I believe this patch belong to homebrew, not fish.

Comment (xyb) [inline Library/Formula/fish.rb]: @mikemcquaid Please do not merge this. I have a discussion with fish maintainer, the `pkg-config` way maybe changed.

Comment (mikemcquaid) [inline Library/Formula/fish.rb]: :ok:

**Comment by DomT4** [comment]
Will take an updated fresh PR whenever something is agreed with upstream. Thank you for your submission to Homebrew!

**Pull request closed without merging**